gson/src/main/java/com/google/gson/internal/bind/MapTypeAdapterFactory.java

```diff
@@ -157,7 +158,9 @@ public Adapter(Gson context, Type keyType, TypeAdapter<K> keyTypeAdapter,
       this.constructor = constructor;
     }
 
-    @Override public Map<K, V> read(JsonReader in) throws IOException {
+    //INSTANCE in #3 may be null as JsonReaderInternalAccess do not initialize it
```

Comment:
This looks like a bug in Gson. Can you produce a test case in which `INSTANCE` is null and the program crashes?

Reply:
The method is defined inside a private class  and is never called so can this method be ruled out as safe?